eos-update-flatpak-repos

```diff
@@ -480,11 +480,134 @@ def _filter_ostree_refs(refs, origin, kind, name='*', prefix=None, arch='*',
+
+    _, content_stream, content_length = OSTree.raw_file_to_content_stream(stream, file_info, None)
+    _, csum_bytes = repo.write_content(None, content_stream, content_length)
+    csum = OSTree.checksum_from_bytes(csum_bytes)
+    mtree.replace_file(file_info.get_name(), csum)
+
+
+def mtree_load_file(repo, mtree, name):
+    files = mtree.get_files()
+    csum = files[name]
+    _, stream, info, _ = repo.load_file(csum)
+    info.set_name(name)
+    return stream, info
+
+
+def mtree_rename_file(repo, mtree, old_name, new_name):
+    stream, info = mtree_load_file(repo, mtree, old_name)
+    info.set_name(new_name)
+    mtree_add_file(repo, mtree, stream, info)
+    mtree.remove(old_name, False)
+
+
+def mtree_load_keyfile(repo, mtree, name):
+    stream, info = mtree_load_file(repo, mtree, name)
+    bytes = stream.read_bytes(info.get_size())
+    keyfile = GLib.KeyFile()
+    keyfile.load_from_bytes(bytes, GLib.KeyFileFlags.KEEP_COMMENTS |
+                                   GLib.KeyFileFlags.KEEP_TRANSLATIONS)
+    return keyfile, info
+
+
+def mtree_add_keyfile(repo, mtree, keyfile, info):
+    data = keyfile.to_data()[0].encode('utf-8')
+    stream = Gio.MemoryInputStream.new_from_data(data)
+    info.set_size(len(data))
+    mtree_add_file(repo, mtree, stream, info)
+
+
+def rename_exports(repo, mtree, old_id, new_id, path=['export']):
+    logging.debug('Renaming files in {}'.format('/'.join(path)))
+
+    files = mtree.get_files()
+    for name,csum in files.items():
+        if name.startswith(old_id):
+            new_name = new_id + name[len(old_id):]
+
+            if path == ['export', 'share', 'applications'] and \
+               name.endswith('.desktop'):
+                logging.info('Rewriting {} as {}'.format(name, new_name))
+                desktop_file, info = mtree_load_keyfile(repo, mtree, name)
+
+                # update the places inside the desktop file where the ID is embedded
+                for key in [GLib.KEY_FILE_DESKTOP_KEY_ICON,
+                            'X-Endless-SplashBackground']:
+                    try:
+                        value = desktop_file.get_string(GLib.KEY_FILE_DESKTOP_GROUP, key)
+                    except GLib.GError as e:
+                        if not e.matches(GLib.KeyFile.error_quark(),
+                                         GLib.KeyFileError.KEY_NOT_FOUND):
+                            raise
+                        continue
+
+                    value = value.replace(old_id, new_id)
+                    desktop_file.set_string(GLib.KEY_FILE_DESKTOP_GROUP, key, value)
+
+                info.set_name(new_name)
+                mtree_add_keyfile(repo, mtree, desktop_file, info)
+                mtree.remove(name, False)
+            elif path == ['export', 'share', 'dbus-1', 'services'] and \
+                 name.endswith('.service'):
+                logging.info('Not changing exported D-Bus service {}'.format(name))
```

Comment:
The eos-build deb-bundled thingies are never DBusActivatable - I don't have a convenient way to check this en masse for the other "artisinal" flatpaks we have, but I also don't see a way to make this work because we can't guarantee the app will actually register the new name we gave it. I suspect given that for all apps, by definition we have the app IDs "wrong", none of them internally use an app ID. To close out this possibility I suggest we just check for and disable DBusActivatable in the .desktop rewriting code.